Move feedback to app header, add mailing list link to home page

pavish opened the pull request:
Fixes #4096
Fixes #4228 
Fixes #4233

## Feedback dropdown
- <img width="1163" alt="Screenshot 2025-02-14 at 3 26 26 PM" src="https://github.com/user-attachments/assets/ce419c31-ed7a-411c-af02-99b2941f991e" />
- <img width="1163" alt="Screenshot 2025-02-14 at 3 26 45 PM" src="https://github.com/user-attachments/assets/0be7ef4a-9699-44c9-b1ea-acb048d0c68e" />

## Resources section - mailing list
- <img width="1067" alt="Screenshot 2025-02-14 at 3 27 17 PM" src="https://github.com/user-attachments/assets/950ee31b-1e2e-4bde-a53e-d141cf927766" />

## Note to reviewers:
- The "Go to Survey" button redirects to 'mathesar.org/survey` which currently does not exist. It is added in https://github.com/mathesar-foundation/mathesar-website/pull/147. 

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. --> 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

pavish commented:
@kgodey @zackkrida I'd like to understand more about how you envision importance given to the feedback option in the product and where or not we should have a single feedback option or separate usecase and general feedback options.

@ghislaineguerin I'd like to discuss the overall UX in placing the feedback in the product and de-cluttering the home page.

zackkrida commented:
@pavish this is great! Some notes:

- I really like where you put the "stay informed" newsletter link and how you replaced the redundancy of the headings/buttons with linked headings. The styling could probably use a small tweak or two from @ghislaineguerin but overall I think it's a really good idea.
- I think the styling of the feedback button gives it a little too much prominence in the UI, and I'd rather see something more like:
    ![image](https://github.com/user-attachments/assets/8ba07d6d-faea-41a2-b54d-5bb8e2d95c66)
    but I'll again defer to @ghislaineguerin there.

As for the use case feedback prominence, I think I'll have to defer to @kgodey there. To me, that should really be part of the onboarding flow where apps commonly ask questions "how big is your company, what competitors do you use, etc." so I'm not sure about where exactly it should be in the interface.

pavish commented:
@zackkrida @ghislaineguerin 

> I think the styling of the feedback button gives it a little too much prominence in the UI

I agree. I would first like to come to an agreement if placing the feedback button on the app header is good, or if we should experiment placing it elsewhere on the product.

I looks like both of you are good with the location and mostly concerned on the visual aspects of it, which we can tweak. Let me know if I that's not the case.

> Is the idea that the feedback will disappear once filled?

No. I envision that it would always the present in the product.

> How do we feel about a general feedback functionality, that remains in place rather than focused just on asking about the use case

Yes, that's the idea. We would only have a general feedback functionality in the location specified in the screenshot. The purpose of this PR was to open up a discussion and I just placed the existing component up there.

@kgodey has expressed opinions that we need two kinds of feedback. A separate use case feedback option somewhere in the product and a general feedback one, and that the use case feedback option should be more prominent.

First, I'd like to come to a decision if we need them to be separate. Second, if yes, I'd like to figure out how and where we'd place both, and how to let the user clearly understand the difference between them.

ghislaineguerin commented:
@pavish I don't think they should be separate, we should figure out how to ask the user a follow up question once they provide feedback. 
As far as location, I do like the one you proposed, I think we could use an icon to make it more noticeable.

zackkrida commented:
To try and summarize a bit, I think the questions for @kgodey are:

1. Confirm: we want to collect **separate** use case feedback, and general product feedback in Mathesar.
2. Confirm: General feedback should be the popup triggered by a button in Mathesar's header
3. Confirm: If we keep the use case feedback, do we:
   - Only show it as part of onboarding when the user has no databases?
   - Show it all the time, below the databases and when no databases are present?
   - Move this feedback form elsewhere in the onboarding process?

kgodey commented:
>     1. Confirm: we want to collect **separate** use case feedback, and general product feedback in Mathesar.

I don't think we necessarily need to do this. I think @ghislaineguerin's idea of asking a follow up question would work. I just don't think it should be the same feedback form as submitting general feedback. We can also replace it with a link to the survey if we update the survey to include a freeform field about the use case (which we should do anyway)

>     2. Confirm: General feedback should be the popup triggered by a button in Mathesar's header

Yes, I like this, it's prominent.

>     3. Confirm: If we keep the use case feedback, do we:        
>        * Only show it as part of onboarding when the user has no databases?
>        * Show it all the time, below the databases and when no databases are present?
>        * Move this feedback form elsewhere in the onboarding process?

Skipping this since it was answered above.

zackkrida commented:
Hi @pavish, just an update on how to proceed here:

1. Update the header feedback form to be a generic feedback form.
2. Remove the use case feedback form entirely from Mathesar. 
3. In the "thank you for your feedback" text displayed after a user submits the generic feedback form, display a link to the survey, with text like: "Thanks for your feedback! If you have a minute, we would love to hear more about how you're using Mathesar in our short survey."
4. The survey has already been updated to include a free-form field about the user's use case and industry.

ghislaineguerin commented:
@pavish I have created the wireframes for the two different options discussed today:
https://www.figma.com/design/jAHxpsgdWUNh6mZoLSmGRm/Feedback-Form?node-id=0-1&t=nCFgGwmO6IUIpP2u-1

@zackkrida I think option 2 is close to what you are describing above.

zackkrida commented:
@ghislaineguerin, yes, Option 2 looks perfect for what I was thinking! Concerning the designs:

I assume we would omit the checkbox about contacting users and the dropdown for types of feedback in this version? Perhaps the dropdown is simple enough to implement, assuming we know what types of feedback we want to support, but what would the email checkbox even do? We don't have email addresses for every user so I'm uncertain how it would work.

pavish commented:
@zackkrida @ghislaineguerin I've simplified the UX and updated the PR description with the screenshots.

I've not included the call team option, I'm not sure if it's something we want to do now and if we have a public call scheduling link. @zackkrida Let me know if we have a link that I can add.

seancolsen reviewed (approved):
Looks good, @pavish.

I haven't actually looked at the submitted feedback to verify that it ends up in our hands. I'll try to spend some time on that maybe today if I can. But I'm skipping that for now because it doesn't seem relevant to this PR.

I pushed a few additional commits. I'll leave this PR for you to merge, assuming you're satisfied with these changes I've added.

These should be straightforward:

- e3bd9c858874c8bca336c94a16573cefbcffebcc
- ffa80f1abcae7fa6a4d72a9038bf7cf23e343c3e

I also pushed 521b6a956a00c9509f8e644fa3afb0f233f448ba which I'd like you to look over. Before this commit, this nav header felt a bit too squished when viewing some pages on narrow viewports. This commit improves the logic used to simplify elements in the nav header. We had previously been applying this sort of logic to the "Shortcuts" menu when that menu was placed in the nav header where the "Feedback" item is now. Here's a quick video which demonstrates the problem and my changes to fix it:

https://github.com/user-attachments/assets/d6c4daf1-0145-4774-ba52-edca736967cd

I'm assigning this back to you, @pavish so that you can look over these changes and merge.

zackkrida commented:
Really nice work @pavish!
